sender.go

```diff
@@ -125,10 +133,12 @@ func (sender *Sender) start() {
 			i += binary.PutUvarint(header[i:], uint64(flags))
 			frameBuffer.Write(header[:i])
 
-			frameEncoder.enableCompression(msg.Compressed())
-			frameEncoder.write(body)
-
-			if msgType := msg.Type(); !msgType.isAck() {
+			if msgType := msg.Type(); msgType.isAck() {
+				// ACKs don't go through the codec nor contain a checksum:
```

Comment:
Any idea why this wasn't caught in any of our testing?  For example the manual testing with the Todo sample app, or the [unit tests](https://github.com/couchbase/sync_gateway/pull/3191) which sends blip requests (via go-blip) to sync gateway.

Asking because I'd like to try to enhance the tests to catch this if possible.  Any hints welcome.

Do the tests just need to send larger multi-frame messages in order to trigger ACK's to be sent?

Reply:
It wasn't caught because (a) go-blip was internally consistent, just inconsistent with the spec and LiteCore, and (b) ACKs only get sent when a message is larger than 50KB.